rust/core-lib/src/tabs.rs

```diff
@@ -491,6 +496,27 @@ impl<W: Write + Send + 'static> Documents<W> {
                 self.plugins.stop_plugin(&view_id, &plugin_name);
                 None
             }
+            PluginRpc  { view_id, receiver, mut rpc } => {
+                // inserting view_id into params here because the plugin needs
+                // to know where this command originates. This feels a bit
+                // clumbsy though. At the very least we should use some obscure
+                // key so we don't collide with any possible user-supplied keys?
+                // Or we could validate keys on plugin load or something.
+                assert!(rpc.params_ref().is_object(), "params must be an object");
+                rpc.params_ref_mut().as_object_mut()
+                    .map(|p| p.insert("view_id".to_owned(), json!(view_id)));
+                match rpc {
+                    PlaceholderRpc::Notification { ref method, ref params } => {
+                        self.plugins.dispatch_notification(&view_id, &receiver,
+                                                           method, params);
+                        None
+                    }
+                    PlaceholderRpc::Request { ref method, ref params } => {
```

Comment:
I agree that in client->plugin RPC requests probably aren't necessary (at least as custom commands; they might be useful for things like requesting completions?) but I wanted to build in the machinery now because I think they _will_ be used for inter-plugin communication.

This does raise the point though that there are some places in this PR where I'm expecting client->plugin RPC requests; I'll clean those up.